packages/x-data-grid-premium/src/components/GridColumnMenuRowUngroupItem.tsx

```diff
@@ -32,7 +33,8 @@ export function GridColumnMenuRowUngroupItem(props: GridColumnMenuItemProps) {
     onClick(event);
   };
 
-  const name = columnsLookup[colDef.field].headerName ?? colDef.field;
+  const headerName = value(columnsLookup[colDef.field].headerName);
+  const name = isString(headerName) ? headerName : colDef.field;
```

Comment:
I wonder why we're using `headerName` here at all. I feel like it should only be `colDef.field`.